src/components/Reports/FourteenMonthReports/FourteenMonthReport.tsx

```diff
@@ -109,148 +100,128 @@ export const FourteenMonthReport: React.FC<Props> = ({
       year: '2-digit',
     });
 
-  const csvData = useMemo(() => {
-    if (!contacts) {
-      return [];
-    }
+  // If there are multiple tables, concatenate them all together
+  const csvData = useMemo(
+    () =>
+      currencyTables.flatMap(({ currency, orderedContacts, totals }) => {
+        // Each table starts with two rows of headers
+        const csvHeaders = [
+          [t('Currency'), currency],
+          [
+            t('Partner'),
+            t('Status'),
+            t('Commitment Amount'),
+            t('Commitment Currency'),
+            t('Commitment Frequency'),
+            t('Committed Monthly Equivalent'),
+            t('In Hand Monthly Equivalent'),
+            t('Missing In Hand Monthly Equivalent'),
+            t('In Hand Special Gifts'),
+            t('In Hand Date Range'),
+            ...totals.map(({ month }) => month),
+            t('Total (last month excluded from total)'),
+          ],
+        ];
 
-    const months =
-      data?.fourteenMonthReport.currencyGroups[0]?.totals.months ?? [];
+        // Then one row for each contact
+        const csvBody = orderedContacts.map((contact) => {
+          const numMonthsForMonthlyEquivalent = Math.max(
+            4,
+            parseInt(contact.pledgeFrequency ?? '4'),
+          );
 
-    const csvHeaders = [
-      [t('Currency'), data?.fourteenMonthReport.salaryCurrency],
-      [
-        t('Partner'),
-        t('Status'),
-        t('Commitment Amount'),
-        t('Commitment Currency'),
-        t('Commitment Frequency'),
-        t('Committed Monthly Equivalent'),
-        t('In Hand Monthly Equivalent'),
-        t('Missing In Hand Monthly Equivalent'),
-        t('In Hand Special Gifts'),
-        t('In Hand Date Range'),
-        ...months.map(({ month }) => month),
-        t('Total (last month excluded from total)'),
-      ],
-    ];
+          const pledgedMonthlyEquivalent =
+            contact.status === 'Partner - Financial' &&
+            contact.pledgeAmount &&
+            contact.pledgeFrequency
+              ? Math.round(
+                  contact.pledgeAmount / parseFloat(contact.pledgeFrequency),
+                )
+              : '';
 
-    const csvBody = [
-      ...contacts.map((contact) => {
-        const numMonthsforMonthlyEquivalent = Math.max(
-          4,
-          parseInt(contact.pledgeFrequency ?? '4'),
-        );
+          const inHandMonths = contact.months?.slice(
+            15 - numMonthsForMonthlyEquivalent - 1,
+            15 - 1,
+          );
 
-        const pledgedMonthlyEquivalent =
-          contact.status === 'Partner - Financial' &&
-          contact.pledgeAmount &&
-          contact.pledgeFrequency
-            ? Math.round(
-                contact.pledgeAmount / parseFloat(contact.pledgeFrequency),
-              )
-            : '';
+          const inHandMonthlyEquivalent =
+            contact.status === 'Partner - Financial' &&
```

Comment:
I think it could be a future improvement if you're OK with that since this isn't new code, just code moved into a separate file. Since this data comes from the REST proxy, it isn't being translated into a ContactStatusEnum automatically for us. It won't break when task phases goes live because that status wasn't renamed, right?

Reply:
That is correct. We could put a TODO in here to refactor later to use the id instead of the display value.

Reply:
Actually, the contact status did change in https://github.com/CruGlobal/mpdx_api/pull/2799, so I'll have to fix that. I think I'll do it in a PR based on this one because this one is just supposed to be about creating one table per currency.

Reply:
#1080 should fix this